plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererAutoTriggerService.kt

```diff
@@ -198,7 +178,12 @@ class CodeWhispererAutoTriggerService : CodeWhispererAutoTriggerHandler, Disposa
         var previousOneAccept: Double = 0.0
         var previousOneReject: Double = 0.0
         var previousOneOther: Double = 0.0
-        val previousOneDecision = CodeWhispererTelemetryService.getInstance().previousUserTriggerDecision
+        val previousOneDecision =
+            if (CodeWhispererFeatureConfigService.getInstance().getNewAutoTriggerUX()) {
+                CodeWhispererTelemetryServiceNew.getInstance().previousUserTriggerDecision
+            } else {
+                CodeWhispererTelemetryService.getInstance().previousUserTriggerDecision
```

Comment:
i feel telemetryService can be shared, and have likely maintain 2 previousUserTriggerDecision
```
class CwsprTelemetryService {
      private val previousUserTriggerDecisionOld: Queue<Decision>
      private val previousUserTrigerDecisionNew: Queue<Decision>
     

     val previousUserTriggerDecision: Decision
               get() {
                      if (newUX) {} else {}
                }
}
```